### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by Kasaderos** [nurtidev/mechta_test#1]
Good, but you can do it better
The result channel buffer depends on numBlocks and you wait all workers then you sum all sums in the main thread.
You shouldn't wait all workers in the main thread, just start sum if the result channel isn't empty, and close channel in a separate goroutine

**Comment by nurtidev** [comment]
You are right. There is no need to keep the main goroutine waiting until all goroutines have completed their work before starting to calculate the total. 
Updated the code with your suggestions, thank u!
